pkg/datapath/connector/pod-link.go

```diff
@@ -0,0 +1,134 @@
+package connector
+
+import (
+	"fmt"
+
+	"github.com/cilium/cilium/pkg/datapath/link"
+	"github.com/cilium/ebpf"
+
+	"github.com/containernetworking/plugins/pkg/ns"
+	"github.com/vishvananda/netlink"
+
+	"golang.org/x/sys/unix"
+)
+
+// SetupNicInRemoteNs creates a tail call map, renames the netdevice inside
+// the target netns and attaches a BPF program to it.
+// egress is on 0. ingress is on 1.
+// NB: Do not close the returned map before it has been pinned. Otherwise,
+// the map will be destroyed.
+func SetupNicInRemoteNs(netNs ns.NetNS, srcIfName, dstIfName string, egress bool, ingress bool) (*ebpf.Map, error) {
+	rl := unix.Rlimit{
+		Cur: unix.RLIM_INFINITY,
+		Max: unix.RLIM_INFINITY,
+	}
+
+	err := unix.Setrlimit(unix.RLIMIT_MEMLOCK, &rl)
+	if err != nil {
+		return nil, fmt.Errorf("unable to increase rlimit: %s", err)
+	}
+
+	m, err := ebpf.NewMap(&ebpf.MapSpec{
+		Type:       ebpf.ProgramArray,
+		KeySize:    4,
+		ValueSize:  4,
+		MaxEntries: 2,
```

Comment:
The map size should probably depend on whether `egress` and `ingress` is set.

Reply:
> I'm wondering whether you could add a CI test to `test/k8sT/DatapathConfiguration.go` which would chain with https://www.cni.dev/plugins/main/ipvlan/ and check basic connectivity? Happy to help with it.

Sure, I'm working for it.

Reply:
I will change it.